lib/IMAP/PreviewEnhancer.php

```diff
@@ -50,9 +61,14 @@ public function __construct(IMAPClientFactory $clientFactory,
 	 * @return Message[]
 	 */
 	public function process(Account $account, Mailbox $mailbox, array $messages): array {
-		$needAnalyze = array_reduce($messages, static function (array $carry, Message $message) {
+		$needAnalyze = array_reduce($messages, function (array $carry, Message $message) {
```

Comment:
The purpose of this `array_reduce` is to find messages that need processing.

Placing the avatar logic in it is convenient but a hack. Please put this somewhere else.

I also believe that the preview enhancer is run for background job and CLI syncs. Doesn't this slow them down while the volatile avatar result is not used? Only REST calls to get the message/thread lists should have this avatar pre-fetching.